src/backend/distributed/commands/function.c

```diff
@@ -187,22 +187,42 @@ create_distributed_function(PG_FUNCTION_ARGS)
 	EnsureFunctionOwner(funcOid);
 
 	ObjectAddressSet(functionAddress, ProcedureRelationId, funcOid);
-	ErrorIfFunctionDependsOnExtension(&functionAddress);
 
 	/*
-	 * when we allow propagation within a transaction block we should make sure to only
-	 * allow this in sequential mode
+	 * If the function is owned by an extension, only update the
+	 * pg_dist_object, and not propagate the CREATE FUNCTION. Function
+	 * will be created by the virtue of the extension creation.
 	 */
-	EnsureSequentialModeForFunctionDDL();
+	if (IsObjectAddressOwnedByExtension(&functionAddress, &extensionAddress))
+	{
+		/* Ensure corresponding extension is in pg_dist_object. */
+		if (!IsObjectDistributed(&extensionAddress))
+		{
+			ereport(ERROR, (errmsg("Extension(%s) owning the "
+								   "function(%s) is not distributed",
+								   get_extension_name(extensionAddress.objectId),
+								   get_func_name(funcOid))));
```

Comment:
Would it be possible to get the extension distributed via the codepath in `EnsureDependenciesExistOnAllNodes`?
I would suspect we could resolve the extension via its dependency line from the function, although I haven't looked deeply into this.

Ideally, from a UX perspective, we would like to automatically distribute the extension if it was not already distributed.

If not already supported I expect we can quite easily make sure that `EnsureDependenciesExistOnAllNodes` would find the function belonging to an extension, mark it as distributed and create the extension on all nodes. All within the same transaction as the user is distributing the function. Meaning, the user experiences it as it-just-works-tm.

Reply:
Thanks. Yes, it appears that EnsureDependenciesExistOnAllNodes() should work.

```
-- Distribute an extension-function where extension is not in pg_dist_object
SELECT create_distributed_function('seg_in(cstring)');
ERROR:  function "seg_in(cstring)" does not exist
```
This is more like a chicken and egg problem; if an extension is not there, in ideal conditions, extension-function shouldn't been there in the first place. I'v tried to create the function and ALTER to DEPEND on EXTENSION, but we already have the code to block it.
`
ERROR:  distrtibuted functions are not allowed to depend on an extension`

Reply:
OK, I've hacked it around by updating  pg_dist_object and it seems to work
![image](https://user-images.githubusercontent.com/44680808/152050505-e22f813d-64ea-428d-9f4a-552002f5fc6c.png)

Reply:
Instead of removing the entry in `citus.pg_dist_object` we can prevent the creation of the extension to propagate to the workers.

The easiest is to `SET citus.enable_ddl_propagation TO false;` before the creation of the extension and `RESET` the GUC right after.

This plays nicely with MX as well.

Reply:
Was looking for such solutions, Thanks for the tip. Changed it now.